sentry-android-core/src/main/java/io/sentry/android/core/DefaultAndroidEventProcessor.java

```diff
@@ -731,6 +733,27 @@ private void setAppPackageInfo(final @NotNull App app, final @NotNull PackageInf
     app.setAppIdentifier(packageInfo.packageName);
     app.setAppVersion(packageInfo.versionName);
     app.setAppBuild(ContextUtils.getVersionCode(packageInfo));
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN) {
+      final Map<String, Object> permissions = new HashMap<>();
+      final String[] requestedPermissions = packageInfo.requestedPermissions;
+      final int[] requestedPermissionsFlags = packageInfo.requestedPermissionsFlags;
+
+      if (requestedPermissions != null
+          && requestedPermissions.length > 0
+          && requestedPermissionsFlags != null
+          && requestedPermissionsFlags.length > 0) {
+        for (int i = 0; i < requestedPermissions.length; i++) {
+          if ((requestedPermissionsFlags[i] & REQUESTED_PERMISSION_GRANTED)
+              == REQUESTED_PERMISSION_GRANTED) {
+            permissions.put(requestedPermissions[i], "granted");
+          } else {
+            permissions.put(requestedPermissions[i], "not_granted");
+          }
+        }
+      }
+      app.setUnknown(permissions);
```

Comment:
Something for TSC?

Reply:
The protocol is flexible enough but sure, please add a new item to TSC and join the next meeting.
Maybe they need to add it to the protocol and we'd need to amend the develop docs https://develop.sentry.dev/sdk/event-payloads/contexts/#app-context

Reply:
Btw a `permissions` map would make more sense to me rather than an entry for each permission, e.g.

```
{
  "contexts": {
    "app": {
      "app_start_time": "2022-05-04T10:01:41.818Z",
      "app_identifier": "io.sentry.samples.android",
      "app_name": "Sentry sample",
      "app_version": "1.1.0",
      "app_build": "2",
      "type": "app",
      "permissions": {
        "android.permission.ACCESS_NETWORK_STATE": "granted",
        "android.permission.CAMERA": "granted",
        "android.permission.FOREGROUND_SERVICE": "granted",
        "android.permission.INTERNET": "granted",
        "android.permission.READ_EXTERNAL_STORAGE": "not_granted",
        "android.permission.READ_PHONE_STATE": "not_granted",
        "android.permission.WRITE_EXTERNAL_STORAGE": "not_granted"
      }
    }
  }
}
```

Reply:
So it looks like this?
<img width="1225" alt="image" src="https://user-images.githubusercontent.com/4999776/166879628-1a04c31e-2b44-4a96-97b7-47d144e1f3a5.png">

I think I'd get rid of `android.permission.` prefix and just keep the permission name, wdyt?
Also, do we then need to change the protocol anyway, as for the user it doesn't look any different than the other fields of the App context? Adding a new field to the protocol would require changes on relay and sentry-frontend as far as I can see.

Reply:
Removing `android.permission` works for me.
the `granted` or `not_granted` could be a simple boolean as well? unless there are more than 2 states.
Afaik the protocol is flexible the render the UI, but relay prefers the have the possible fields mapped in their code, docs and develop docs, and frontend, but please bring it to TSC.

Reply:
I think there's technically `denied`, `granted` and `not_asked`, but I wasn't able to find out how to determine `not_asked`. I will try again and switch to booleans in case I fail.

Alright, will add it to the TSC. Though it looks like a no-brainer, so I probably could add it everywhere myself, but let's see.

Reply:
Please align the values with @brustolin and @philipphofmann 
Or define the values and add them as a comment to https://github.com/getsentry/sentry-cocoa/issues/1014
The goal is that if one day, this gets indexed, the discover queries should work for both SDKs.

Reply:
mmm, I think it's gonna be hard to align 100%, cause the permissions diverge between the platforms. But I agree, the ones that match we should align 👍

Reply:
I meant the data structure, either a boolean or String as value, and if String, the values should be the same on Android and iOS, e.g. `not_asked`, I'd not like to be `notAsked` on iOS, discover queries wouldn't work.

Reply:
@brustolin, can you align with @romtsn here as you are currently investigating the same feature on iOS, please?

Reply:
In iOS we have Denied, Granted, notDetermined and Restricted. 
I think we just need to add `restricted` and 'not determined' permissions should not be in the event.

Reply:
Alright, I've changed it to booleans, but I can change it back to `granted`/`denied` to align. Even though we cannot determine whether `denied` was actually denied or just not yet asked.

Reply:
Also, not every permission uses the `restricted` option, even though all of them contains the enum value

Reply:
Hey @romtsn did you changed it back already? Maybe we shouldn't. I just realize that `restricted` in iOS means not available because some configuration, which basically means denied.

Reply:
I didn't. Let's keep booleans then.

Reply:
I think permissions could be just a string array, like this:

```json
{
  "contexts": {
    "app": {
      "app_start_time": "2022-05-04T10:01:41.818Z",
      "app_identifier": "io.sentry.samples.android",
      "app_name": "Sentry sample",
      "app_version": "1.1.0",
      "app_build": "2",
      "type": "app",
      "permissions": [
        "android.permission.ACCESS_NETWORK_STATE",
        "android.permission.CAMERA",
        "android.permission.FOREGROUND_SERVICE",
        "android.permission.INTERNET",
        "android.permission.READ_EXTERNAL_STORAGE",
        "android.permission.READ_PHONE_STATE",
        "android.permission.WRITE_EXTERNAL_STORAGE"
      ]
    }
  }
}
```

Reply:
But this is a list of granted permissions then? Not sure if it's explicit enough for the user. But we can discuss - I added this topic to the next TSC